File: solutions/861.cpp
class Solution {
public:
    int matrixScore(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        for (int i = 0; i < m; ++i) {
            if (grid[i][0] == 0) {
                for (int j = 0; j < n; ++j) {
                    grid[i][j] = grid[i][j] ^ 1;
                }
            }
        }
        
        int res = 0;
        for (int j = 0; j < n; ++j) {
            int cnt = 0;
            for (int i = 0; i < m; ++i) {
                if (grid[i][j] == 1) cnt++;
            }
            cnt = max(cnt, m - cnt);
            res += cnt * (1 << (n - 1- j));
        }
        return res;
    }
};

File: rust_solutions/861.rs
impl Solution {
    pub fn matrix_score(mut grid: Vec<Vec<i32>>) -> i32 {
        let m = grid.len();
        let n = grid[0].len();
        for i in 0..m {
            if (grid[i][0] == 0) {
                for j in 0..n {
                    grid[i][j] ^= 1;
                }
            }
        }
        let mut res = 0;
        for j in 0..n {
            let mut cnt = 0;
            for i in 0..m {
                if grid[i][j] == 1 {
                    cnt += 1;
                }
            }
            cnt = cnt.max(m - cnt);
            res += cnt * (1 << (n - 1 - j));
        }
        return res as i32;
    }
}
